src/theme/components/hero.ts

```diff
@@ -0,0 +1,247 @@
+// From the `wrapper` SCSS mixin.
+// TODO: Eventually we may be able to put shared JS style objects
+// into a utils file for other mixins that are shared.
+const wrapperStyles = {
+  marginY: 0,
+  marginX: "auto",
+  maxWidth: "1280px",
+  paddingTop: 0,
+  paddingBottom: 0,
+  paddingRight: 0,
+  paddingLeft: 0,
+  width: "100%",
+};
+// TODO: Is this being used?
+const placeholder = {
+  "+ .placeholder": {
+    flex: "0 1 200px",
+    marginLeft: 12, // --space-xl
+  },
+};
+
+const variants = {
+  primary: {
+    alignItems: "center",
+    display: "flex",
+    flexFlow: {
+      base: "column nowrap",
+      md: "row nowrap",
+    },
+    justifyContent: "center",
+    minHeight: "350px",
+    content: {
+      bg: "ui.black",
+      color: "ui.white",
+      flex: {
+        base: "0 0 100%",
+        md: "0 0 60%",
+      },
+      maxWidth: { md: "960px" },
+      paddingTop: 16, // --space-xxl
+      paddingBottom: 16, // --space-xxl
+      paddingRight: 8, // --space-l
+      paddingLeft: 8, // --space-l
+      a: {
+        color: "inherit",
+        display: "inline-block",
+      },
+      heading: {
+        marginBottom: 0,
+      },
+      bodyText: {
+        marginBottom: 0,
+      },
+      ...placeholder,
+    },
+  },
+  secondary: {
+    overflowX: "hidden",
+    content: {
+      ...wrapperStyles,
+      paddingRight: 4, // --space-s
+      paddingLeft: 4, // --space-s
+      display: "flex",
+      flexFlow: {
+        base: "column nowrap",
+        md: "row wrap",
+      },
+      img: {
+        flex: {
+          base: "1 1 100%",
+          md: "0 0 250px",
+        },
+        order: { md: "3" },
+        height: "150px",
+        minWidth: "0", // https://github.com/philipwalton/flexbugs/issues/41
+        objectFit: "cover",
+        width: "100%",
+      },
+    },
+    heading: {
+      marginBottom: 0,
+      bg: "ui.black",
+      color: "ui.white",
+      flex: "1 1 100%",
+      marginTop: 0,
+      paddingBottom: 1, // --space-xxs
+      position: "relative",
+      zIndex: 0,
+      order: { md: "1" },
+      _before: {
+        bg: "ui.black",
+        content: `""`,
+        height: "100%",
+        left: "-2000px",
+        position: "absolute",
+        width: "4000px",
+        zIndex: "-1",
+      },
+      ".nypl--books-and-more &": {
```

Comment:
What exactly does this do?

Reply:
This is for the cases where a page has a specific class and the `Hero` needs to reflect that page. So something like:

```jsx
<div className="nypl--books-and-more">
  <Hero {...props} />
</div>
```

This CSS rule will take effect for this pattern. We do, however, need to update this and not follow this moving forward, but maybe it's too quick of a move to do this now? It can part of the migration process. If we do need to make this move now, these styles can be variants rather than relying on the parent's class name.

So the above would be:

```jsx
<div >
  <Hero type={HeroTypes.BooksAndMore} {...props} />
</div>
```

Reply:
Right.  I think this was a nice idea, but it was never fully fleshed out and implemented throughout DS.  I would like to move away from this and make the color assignment more of a manual props process.

Reply:
Removed the parent wrapper class names in favor of additional types.